backend/consensus/base.py

```diff
@@ -471,6 +518,44 @@ def dispatch_transaction_status_update(
             )
         )
 
+    def run_appeal_window_loop(self):
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        loop.run_until_complete(self._appeal_window())
+        loop.close()
+
+    async def _appeal_window(self):
+        FINALITY_WINDOW = int(os.getenv("FINALITY_WINDOW"))
+        while True:
+            with self.get_session() as session:
+                chain_snapshot = ChainSnapshot(session)
+                transactions_processor = TransactionsProcessor(session)
+                accepted_transactions = chain_snapshot.get_accepted_transactions()
+                for transaction in accepted_transactions:
+                    transaction = transaction_from_dict(transaction)
+                    if not transaction.appealed:
+                        if (
+                            int(time.time()) - transaction.timestamp_accepted
+                        ) > FINALITY_WINDOW:
+                            self.finalize_transaction(
+                                transaction,
+                                transactions_processor,
+                                lambda contract_address, session=session: ContractSnapshot(
+                                    contract_address, session
+                                ),
+                            )
+                            session.commit()
+                    else:
+                        transactions_processor.set_transaction_appeal(
```

Comment:
@kstroobants could you please add some comments here to explain the logic? for example, why do you set the Appeal as false when the transaction was appealed?

Reply:
I added a lot of comments to this file in this commit https://github.com/yeagerai/genlayer-studio/pull/637/commits/27624bbf7bb6a3935352e45c89160d6c0428e06a.

Yes, it did not make sense to put it on False. It is solved in next PRs.